test/integration/test_server.cpp

```diff
@@ -1870,6 +1927,130 @@ TEST_CASE("WriteBatchHistory upsert and delete", "[integration][write-batch]") {
   }
 }
 
+TEST_CASE("WriteBatchHistory transaction history upsert",
```

Comment:
This tests both WriteBatch and WriteBatchHistory together, interspersed? Maybe adapt the name, so it's obvious why WriteBatch is there.